app/actions/local/file.ts

```diff
@@ -34,3 +38,33 @@ export const updateLocalFilePath = async (serverUrl: string, fileId: string, loc
         return {error};
     }
 };
+
+export const getAllFilesInCachesDirectory = async () => {
+    try {
+        const appDatabase = DatabaseManager.appDatabase;
+        const servers = await queryAllServers(appDatabase!.database);
+        if (!servers.length) {
+            return {error: 'No servers'};
+        }
+
+        const cachesDirectories = servers.map((server) => `${FileSystem.CachesDirectoryPath}/${hashCode(server.url)}`);
+
+        const files: FileSystem.ReadDirItem[][] = [];
+
+        for await (const directory of cachesDirectories) {
+            const directoryFiles = await FileSystem.readDir(directory);
+            files.push(directoryFiles);
+        }
+        const flattenedFiles = files.flat();
+        const totalSize = flattenedFiles.reduce((acc, file) => acc + file.size, 0);
+
+        return {
+            files: flattenedFiles,
+            totalSize,
+        };
+    } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log('Failed getAllFilesInCachesDirectory', error);
+        return {error};
+    }
```

Comment:
see utils/file/index.ts there is already a function `deleteFileCache(serverUrl: string)` basically we should have a function there called `getFilesInCacheDirectory(serverUrl: string)` that does this and have the local action call the utility function, in fact it may be not even an action

Reply:
refactored

Reply:
I don't see this refactor